Tools/Scripts/webkitpy/common/net/bugzilla/results_fetcher.py

```diff
@@ -207,3 +207,28 @@ def lookup_ews_results_from_bugzilla(
         raise RuntimeError(msg)
 
     return lookup_ews_results_from_bugzilla_patch(patch, bot_filter_name)
+
+
+def lookup_ews_results_from_pr(pr):
+    urls = [status.url for status in pr.statuses]
+    return lookup_ews_results(urls)
+
+
+def lookup_ews_results_from_repo_via_pr(repository):
+    source_remote = repository.default_remote
+    remote_repo = repository.remote(name=source_remote)
+    if not remote_repo.pull_requests:
+        msg = "Remote repo doesn't support pull requests"
```

Comment:
Why not just `raise ValueError` directly?

Reply:
see https://docs.astral.sh/ruff/rules/raw-string-in-exception/; essentially it avoids the string being duplicated in the traceback, which makes it somewhat harder to read